lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVMBase.h

```diff
@@ -529,7 +529,7 @@ class ConvertTritonGPUOpToLLVMPatternBase {
     assert(rank > 0);
     SmallVector<Value> multiDim(rank);
     if (rank == 1) {
-      multiDim[0] = linear;
+      multiDim[0] = urem(linear, i32_val(shape[0]));
```

Comment:
The given value could be greater than the shape

Reply:
I believe this could be guarded statically to avoid the cost of the remainder when not necessary?

Reply:
You are definitely right, my bad